### PR TITLE
Misc docstring fixes for cosmology

### PR DESCRIPTION
**Pull request by dstansby**
### Description

I discovered https://pypi.org/project/velin/, which is a small command line utility to automatically fix small/common numpydoc issues in docstrings. I've run it on `astropy`, and it seemed to catch a few things. This PR is fixes to `cosmology`.

- I've checked all these by eye to make sure they look sensible
- I figured it would be best to open one PR/sub-module

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/cosmology/funcs.py [hunk 1]
```diff
@@ -120,59 +120,59 @@ def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500,
...
-        bracket search. For non-monotonic functions such as angular diameter
-        distance this may be used to start the search on the desired side of
-        the maximum, but see Examples below for usage notes.
+         For methods 'Brent' and 'Golden', ``bracket`` defines the bracketing
+         interval and can either have three items (z1, z2, z3) so that
+         z1 < z2 < z3 and ``func(z2) < func (z1), func(z3)`` or two items z1
+         and z3 which are assumed to be a starting interval for a downhill
+         bracket search. For non-monotonic functions such as angular diameter
+         distance this may be used to start the search on the desired side of
+         the maximum, but see Examples below for usage notes.
```
Comment (nstarman) [inline astropy/cosmology/funcs.py:162]: This looks like it's now 5 spaces, not 4. I think this is a mistake!

**Review by nstarman — CHANGES_REQUESTED** [review]
Thanks @dstansby. Looks good except for one Parameter, which is getting over-indented.

File: astropy/cosmology/funcs.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/funcs.py:162]: Reading https://numpydoc.readthedocs.io/en/latest/format.html, all indents are 4 spaces.

Comment (dstansby) [reply astropy/cosmology/funcs.py]: Thanks for catching - should be fixed now.

**Review by nstarman — APPROVED** [review]
Thanks @dstansby!

**Comment by nstarman** [comment]
I think all failures are unrelated and RTD passes.

**Comment by pllim** [comment]
I don't see these errors in `main`... Gosh, why can't Monday start with all green CI?

**Comment by pllim** [comment]
Anyways, I agree failure is unrelated. I don't see how adding whitespace can lead to ResourceWarning. Merging. Thanks!

**Comment by pllim** [comment]
Are you sure this needs backport? If, not please change the milestone and remove the backport-related labels. Thanks!

**Pull request merged**